### ISSUE TITLE
Multiple tests fail on Ubuntu 18.04

### ISSUE THREAD

**Issue by ehkropf** [ehkropf/ModifiedSchwarz#25]
```
[100%] Built target testRunner
Scanning dependencies of target check
-----========== Polynomial ==========-----
  --> Polynomial values ... 
      DONE
  --> Unit Circle Domain check ... 
  --> Points ... 
      DONE
  --> Inside domain ... 
      DONE
  --> On boundary ... 
      DONE
-----========== BoundaryValue check ==========-----
  --> Size check ... 
      DONE
  --> Value check ... 
      DONE
-----========== Real Interpolant ==========-----
  --> Real interpolation ... 
/home/ehkropf/src/ModifiedSchwarz/test/checkInterpolant.cpp:64:1: error: Failure in RealInterp: approx_equal(gi(zb), h(zb), "absdiff", 10.*eps2pi)
      DONE
  --> Real interp derivatives ... 
Test needs written.
/home/ehkropf/src/ModifiedSchwarz/test/checkInterpolant.cpp:76:1: error: Failure in RealInterpD: 0
  --> Complex interpolation ... 
      DONE
  --> Complex interp derivatives ... 
Test needs written.
/home/ehkropf/src/ModifiedSchwarz/test/checkInterpolant.cpp:97:1: error: Failure in ComplexInterpD: 0
-----========== FMM evaluation ==========-----
  --> Result check ... 
      DONE
-----========== Cauchy interpolation ==========-----
  --> Check interpolation ... 
      DONE
-----========== Closure interpolant ==========-----
  --> Points on the boundary ... 
      DONE
  --> Points in the interior ... 
/home/ehkropf/src/ModifiedSchwarz/test/checkClosureInterpolant.cpp:69:1: error: Failure in InteriorInterp: arma::approx_equal(f(interior_points), g(interior_points), "absdiff", 1e-10)
      DONE
  --> Points on boundary and interior ... 
/home/ehkropf/src/ModifiedSchwarz/test/checkClosureInterpolant.cpp:82:1: error: Failure in CombinedInterp: arma::approx_equal(f(zz), g(zz), "absdiff", 1e-10)
      DONE
-----========== Spectral Constants ==========-----
  --> Trapezoidal rule points ... 
      DONE
  --> Truncation ... 
      DONE
-----========== Spectral data check ==========-----
  --> Matrix ... 
      DONE
  --> Data sharing ... 
      DONE
-----========== Spectral method check ==========-----
  --> Right hand side ... 
/home/ehkropf/src/ModifiedSchwarz/test/checkSpectralMethod.cpp:69:1: error: Failure in TestRHS: approx_equal(rhs.head_rows(rhs.n_rows/2), refRHS, "absdiff", 1e-4)
      DONE
  --> Residual ... 
/home/ehkropf/src/ModifiedSchwarz/test/checkSpectralMethod.cpp:85:1: error: Failure in TestResidual: max(abs(res)/abs(rhs)) < 100*eps2pi
      DONE
  --> Solution ... 
/home/ehkropf/src/ModifiedSchwarz/test/checkSpectralMethod.cpp:101:1: error: Failure in TestSolution: approx_equal(polesInHoles(cx_vec(vectorise(bv)), domain), sol(vectorise(bv)), "absdiff", 1e-6)
      DONE
-----========== Problem check ==========-----
  --> Basic solve ... 
      DONE
-----========== Solution check ==========-----
  --> Evaluate solution ... 
/home/ehkropf/src/ModifiedSchwarz/test/checkSolution.cpp:63:1: error: Failure in EvalSolution: approx_equal(polesInHoles(cx_vec(vectorise(zb)), domain), sol(vectorise(zb)), "reldiff", 10*eps2pi)
      DONE
FAILURE: 9 out of 36 tests failed (9 failures).

**Comment by ehkropf** [comment]
Ok, so the problem is with `arma::fft`. Getting correct results on mac, and incorrect results in 18.04. Why?

**Comment by ehkropf** [comment]
Annoyingly, [test DFT interpolants](https://github.com/ehkropf/Scratchpads/blob/master/cpp/armacheck/fftcheck.cpp) run with the same output on both Mac and 18.04, so it's probably not the `arma::fft`.

**Comment by ehkropf** [comment]
Oddly, the first two provided points on the boundary circle are incorrectly evaluated when `BoundaryValues` is constructed on Linux, but it's fine on Mac.

**Comment by ehkropf** [comment]
In the Armadillo README, it clearly states that

> Caveat: use of the C++11 "auto" keyword is not recommended with Armadillo
objects and expressions. Armadillo has a template meta-programming framework
which creates lots of short lived temporaries that are not handled by auto.

Have cleaned up this use for Armadillo stuff. Should be all fixed by commit 717336f6. (Dev branch.)
